File: hw/maisincho/u3/HW22Strategy/StrategyPy/model/BubbleSort.py
from typing import List
from model.SortingStrategy import SortingStrategy  

class BubbleSort(SortingStrategy):
    def sort(self, array: List[int]) -> None:
        n = len(array)
        for i in range(n - 1):
            for j in range(n - i - 1):
                if array[j] > array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]

File: exams/jacome/U3/ExamU3/model/BubbleSort.js
import { SortingStrategy } from './SortingStrategy.js';

export class BubbleSort extends SortingStrategy {
  sort(arr) {
    const a = [...arr];
    for (let i = 0; i < a.length - 1; i++) {
      for (let j = 0; j < a.length - 1 - i; j++) {
        if (a[j] > a[j + 1]) {
          [a[j], a[j + 1]] = [a[j + 1], a[j]];
        }
      }
    }
    return a;
  }
}
